File: client-module-server/src/main/java/br/edu/les/module/client/domain/TipoUsuario.java
package br.edu.les.module.client.domain;

import br.edu.les.module.client.domain.interfaces.IEntidade;
import br.edu.les.module.client.domain.interfaces.ITipoEntidade;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;

import javax.persistence.*;
import java.io.Serializable;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Component
@Entity
@Table(name = "tipo_usuario")
public class TipoUsuario implements ITipoEntidade {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "tus_id")
	private Integer id;

	@Column(name = "tus_nome_tipo")
	private String tipo;
}

File: client-module-server-node/src/models/TipoUsuario.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from "typeorm";
import { Usuario } from './Usuario';

@Entity({name: "tipo_usuario"})
class TipoUsuario {

    @PrimaryGeneratedColumn({name: "tus_id"})
    id: number;

    @Column({name: "tus_nome_tipo"})
    tipo: string;

    @OneToMany(type => Usuario, tipoCliente => TipoUsuario)
    usuarios: Usuario[];
}

export { TipoUsuario }